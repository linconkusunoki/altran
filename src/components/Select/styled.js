import styled from 'styled-components'

export const Wrapper = styled.div`
  flex: 1;
  margin-top: 15px;

  @media (min-width: 768px) {
    margin-top: 0;
  }
`

export const Select = styled.select`
  display: block;
  padding: 14px;
  border: 1px solid #ddd;
  border-radius: 6px;
  width: 100%;
  background: #f1edff;
`

export const Label = styled.label`
  white-space: nowrap;
`
