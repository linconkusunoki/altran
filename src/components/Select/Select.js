import React from 'react'

import * as S from './styled'

const Select = ({ title, options, value, ...rest }) => {
  return (
    <S.Wrapper>
      <S.Label>
        {title}
        <S.Select {...rest} value={value || ''}>
          <option value="">All</option>
          {options.map((option, index) => (
            <option key={index} value={option}>
              {option}
            </option>
          ))}
        </S.Select>
      </S.Label>
    </S.Wrapper>
  )
}

export default Select
