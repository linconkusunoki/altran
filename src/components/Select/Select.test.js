import React from 'react'
import { render } from '@testing-library/react'
import Select from './'

it('should render correctly', async () => {
  const options = [1, 2, 3]
  const { container } = render(
    <Select title="test" options={options} value="1" onChange={() => {}} />,
  )
  expect(container.firstChild).toMatchSnapshot()
})

it('should render with empty value', async () => {
  const options = [1, 2, 3]
  const { container } = render(
    <Select title="test" options={options} onChange={() => {}} />,
  )
  expect(container.firstChild).toMatchSnapshot()
})
