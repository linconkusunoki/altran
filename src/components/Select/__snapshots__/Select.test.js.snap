// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  class="sc-bdVaJa iUbJjP"
>
  <select
    class="sc-bwzfXH fiTiQi"
  >
    <option
      value=""
    >
      test
    </option>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
  </select>
</div>
`;

exports[`should render with empty value 1`] = `
<div
  class="sc-bdVaJa iUbJjP"
>
  <select
    class="sc-bwzfXH fiTiQi"
  >
    <option
      value=""
    >
      test
    </option>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
  </select>
</div>
`;
